src/common/walFilter/greenplumCommon.h

```diff
@@ -0,0 +1,105 @@
+#ifndef PGBACKREST_GREENPLUMCOMMON_H
+#define PGBACKREST_GREENPLUMCOMMON_H
+
+#include <stdint.h>
+#include "postgres/interface/static.vendor.h"
+
+// Common macros and constants for different versions of Greenplum
+
+// types
+typedef uint32_t TimeLineID;
+typedef uint64_t XLogRecPtr;
+typedef uint8_t RmgrId;
+typedef uint32_t pg_crc32;
+typedef uint32_t TransactionId;
+typedef uint32_t BlockNumber;
+
+// constants
+#define XLOG_BLCKSZ 32768
+/* This flag indicates a "long" page header */
+#define XLP_LONG_HEADER             0x0002
+#define MAXIMUM_ALIGNOF 8
+#define BLCKSZ 32768
```

Comment:
Greenplum [may be configured](https://github.com/arenadata/gpdb/blob/2ce272a173b3353bcb45723ed8b9aa887025f18d/configure#L3724-L3759) with different sizes